### PR TITLE
Add easyLZMA 0.0.7

### PR DESCRIPTION
**Pull request by ericriff**
Specify library name and version:  **easyLZMA/0.07**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericriff** [comment]
I have a few questions for the community regarding this library:
1- The `CMakeLists` doesn't allow us to chose between `static` and `shared` (and it builds both) so I removed the option. Should I put it back and delete unwanted artifacts? Or maybe only copy the right ones during the `package` step?
2- It compiles on Windows but when the `test_package` is ran I get `Error 3221225781`. I have very little experience with windows so I don't know how to fix it. According to the documentation the library is Windows-compatible.
3- The official sources have memory leak issues. It's reported on the issue tracker and there are two PRs opened by the community to fix this, but the main developer is not merging them, I think the project is no longer being developed. I transformed this PRs into a patch. Is this OK?

**Comment by uilianries** [comment]
> 1- The CMakeLists doesn't allow us to chose between static and shared (and it builds both) so I removed the option. Should I put it back and delete unwanted artifacts? Or maybe only copy the right ones during the package step?

Yes, you should put back and delete the unwanted artifacts. Well observed.

> 2- It compiles on Windows but when the test_package is ran I get Error 3221225781. I have very little experience with windows so I don't know how to fix it. According to the documentation the library is Windows-compatible.

That error is related to DLL loading error. Some library is missing in your system. 

> 3- The official sources have memory leak issues. It's reported on the issue tracker and there are two PRs opened by the community to fix this, but the main developer is not merging them, I think the project is no longer being developed. I transformed this PRs into a patch. Is this OK?

If it's no longer developed, I see no problem adding some patches. But the same patches should be in PR to the author, even if the project is abandoned. Other developers will have access to your work, so you will help the community.

> 4- The library doesn't have a license file, I created a question in the conan-index issue tracker and the community suggested to create a file with an extraction from the readme. The hook is not happy about me exporting a license.

Do not extract it to a file and export. Instead, load the file, copy the license content and save to a license file: https://docs.conan.io/en/latest/howtos/extract_licenses_from_headers.html

File: recipes/easylzma/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    default_options = {"fPIC": True}
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
```
Comment (uilianries) [inline recipes/easylzma/all/conanfile.py]:
```suggestion
           del self.options.fPIC
```

Same effect, but more pythonic

File: recipes/easylzma/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self.source_folder)
+        self.copy("*.so*", dst="lib", src=os.path.join(self._source_subfolder,self.name + "-" + self.version,"lib"))
```
Comment (uilianries) [inline recipes/easylzma/all/conanfile.py]: Use zlib as reference for copying the right extension, based in the option "shared": https://github.com/conan-io/conan-center-index/blob/master/recipes/zlib/1.2.11/conanfile.py#L149

File: recipes/easylzma/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#include <iostream>
```
Comment (uilianries) [inline recipes/easylzma/all/test_package/test_package.cpp]: You can rename this file to test_package.c. As easyLZMA is a C project, you can test using a C code too.

File: recipes/easylzma/all/test_package/test_package.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    std::cout << "EasyLZMA test package" << std::endl;
+
+    int rc;
+    int inLen = 10;
+    elzma_file_format format = ELZMA_lzma;
+    elzma_compress_handle hand;
+
+    /* allocate compression handle */
+    hand = elzma_compress_alloc();
+    assert(hand != NULL);
```
Comment (uilianries) [inline recipes/easylzma/all/test_package/test_package.cpp]: assert doesn't work for release build. Use a regular "if" instead.

**Comment by ericriff** [comment]
@uilianries I changed the recipes according to your suggestions, regardless, it still fails for Windows+Static when conan runs the test package, with error 3221225781. The run succeeds if during the package step I also copy the `.dll` file to the bin folder. I'm not really sure why, I'll take a look tomorrow. If you have any suggestions let me know.
Thanks!

**Comment by madebr** [comment]
@ericriff 
See https://github.com/ericriff/conan-center-index/pull/3

**Comment by ericriff** [comment]
@madebr I merged your PR with some minor tweaks, but the same thing keeps happening. I'm still able to build locally.
From your PR the only meaningful change is [9b6e88e](https://github.com/conan-io/conan-center-index/pull/1097/commits/9b6e88ed0a33047e67fed99357377ea4f3a856e6) but I'm not sure if that's an issue.

**Comment by madebr** [comment]
I can reproduce when running:
```
conan create . easylzma/0.0.7@  -s build_type=Debug -s compiler.runtime=MTd
```

**Comment by uilianries** [comment]
@ericriff @madebr Thanks a lot for this great work! :muscle:

**Comment by madebr** [comment]
@ericriff 
See https://github.com/ericriff/conan-center-index/pull/4
The cmake script was overriding the `CMAKE_C_FLAGS`, so `/MTd`  was removed.

**Comment by ericriff** [comment]
> @ericriff
> See [ericriff#4](https://github.com/ericriff/conan-center-index/pull/4)
> The cmake script was overriding the `CMAKE_C_FLAGS`, so `/MTd` was removed.

Thanks for the help!

File: recipes/easylzma/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _license_text(self):
+        # Extract the License/s from the README to a file
+        tmp = tools.load("source_subfolder/README")
```
Comment (uilianries) [inline recipes/easylzma/all/conanfile.py]:
```suggestion
        tmp = tools.load(os.path.join("source_subfolder", "README"))
```

**Comment by dgoel** [comment]
Could we get this PR merged? Seems like it is ready to go and all the comments have been addressed. Thanks!

**Comment by madebr** [comment]
@danimtb Can you take a look?

**Pull request merged**